packages/block-library/src/navigation/edit/index.js

```diff
@@ -513,6 +564,9 @@ function Navigation( {
 								onSelect={ ( { id } ) => {
 									setRef( id );
 								} }
+								onSelectClassic={ ( classicMenu ) => {
```

Comment:
Alternative to a separate `onSelectClassic` would be to detect whether it's a classic menu that's been selected by looking for a property of a classic menu that isn't present on a Navigation Menu (if there is one).

Reply:
I've gone with the approach of detecting the classic menu by checking for the rather obscure `auto_add` prop.